### ISSUE TITLE
Can't forward pass conv2d with kernel_size=1, and padding=1

### ISSUE THREAD

**Issue by thipokKub** [pytorch/pytorch#72591]
### 🐛 Describe the bug

I was trying to implement resnet but using `nn.functional` module instead, and I ran across this error. This behaviour only happen when `kernel_size=1`, and `padding=1`. Is this an expected behaviour?

Reproduce
```python
import torch
from torch.nn import functional as F

F.conv2d(
    torch.rand(16, 3, 10, 10),
    torch.rand(5, 3, 1, 1),
    bias=None,
    stride=1,
    padding=1,
    dilation=1,
    groups=1,
)
```

**Expected Behaviour**
The output is return

**Current Behaviour**
Throw runtime error
```python
RuntimeError: NNPACK SpatialConvolution_updateOutput failed
```

### Versions

PyTorch version: 1.10.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.1 (arm64)
GCC version: Could not collect
Clang version: 13.0.0 (clang-1300.0.29.30)
CMake version: version 3.22.2
Libc version: N/A

Python version: 3.9.9 | packaged by conda-forge | (main, Dec 20 2021, 02:41:07)  [Clang 11.1.0 ] (64-bit runtime)
Python platform: macOS-12.1-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] pytorch-lightning==1.5.7
[pip3] pytorchcv==0.0.67
[pip3] torch==1.10.0
[pip3] torch-summary==1.4.5
[pip3] torchmetrics==0.6.2
[pip3] torchvision==0.11.2
[conda] numpy                     1.19.5           py39h1f3b974_2    conda-forge
[conda] pytorch                   1.10.0          cpu_py39h7601aee_0    conda-forge
[conda] pytorch-cpu               1.10.0          cpu_py39hd610c6a_0    conda-forge
[conda] pytorch-lightning         1.5.7              pyhd8ed1ab_0    conda-forge
[conda] pytorchcv                 0.0.67                   pypi_0    pypi
[conda] torch-summary             1.4.5                    pypi_0    pypi
[conda] torchmetrics              0.6.2              pyhd8ed1ab_0    conda-forge
[conda] torchvision               0.11.2                   pypi_0    pypi

**Comment by jbschlosser** [comment]
Hey @thipokKub, thanks for the report! Do you mind trying this on a master version of PyTorch? I wasn't able to reproduce the problem, and a decent amount has changed in that area since 1.10.0.

**Comment by thipokKub** [comment]
**TL;DR** The error happens when installed `torch` package from `conda` only, but `pip` wheel seems to be working just fine. However, I only tested with `osx_aarch64`, and not sure on other platform

Hello, to clarify, my machine is macbook M1x (aarch64) with the pytorch library I installed is from [conda-forge/miniforge](https://github.com/conda-forge/miniforge), which then install pytorch using the following command `conda install pytorch torchvision -c pytorch`

I believe that _trying on master_ meant compiling from source, but it failed producing the following error

```
$> export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
$> MACOSX_DEPLOYMENT_TARGET=10.9 CC=clang CXX=clang++ python setup.py install
Building wheel torch-1.12.0a0+git81a1330
-- Building version 1.12.0a0+git81a1330
cmake --build . --target install --config Release
[1/2] Linking CXX shared library lib/libcaffe2_detectron_ops.dylib
FAILED: lib/libcaffe2_detectron_ops.dylib 
: && /Library/Developer/CommandLineTools/usr/bin/clang++ -falign-functions=8 -Wno-deprecated -fvisibility-inlines-hidden -Wno-deprecated-declarations -DUSE_PTHREADPOOL -Xpreprocessor -fopenmp -I/Users/thipok.tham/miniforge3/envs/pytorch_build/include -DNDEBUG -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-range-loop-analysis -Wno-pass-failed -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -Wno-invalid-partial-specialization -Wno-typedef-redefinition -Wno-unknown-warning-option -Wno-unused-private-field -Wno-inconsistent-missing-override -Wno-aligned-allocation-unavailable -Wno-c++14-extensions -Wno-constexpr-not-const -Wno-missing-braces -Qunused-arguments -fcolor-diagnostics -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-unused-private-field -Wno-missing-braces -Wno-c++14-extensions -Wno-constexpr-not-const -O3 -DNDEBUG -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -mmacosx-version-min=10.9 -dynamiclib -Wl,-headerpad_max_install_names -rdynamic -o lib/libcaffe2_detectron_ops.dylib -install_name @rpath/libcaffe2_detectron_ops.dylib modules/detectron/CMakeFiles/caffe2_detectron_ops.dir/group_spatial_softmax_op.cc.o modules/detectron/CMakeFiles/caffe2_detectron_ops.dir/ps_roi_pool_op.cc.o modules/detectron/CMakeFiles/caffe2_detectron_ops.dir/roi_pool_f_op.cc.o modules/detectron/CMakeFiles/caffe2_detectron_ops.dir/sample_as_op.cc.o modules/detectron/CMakeFiles/caffe2_detectron_ops.dir/select_smooth_l1_loss_op.cc.o modules/detectron/CMakeFiles/caffe2_detectron_ops.dir/sigmoid_cross_entropy_loss_op.cc.o modules/detectron/CMakeFiles/caffe2_detectron_ops.dir/sigmoid_focal_loss_op.cc.o modules/detectron/CMakeFiles/caffe2_detectron_ops.dir/smooth_l1_loss_op.cc.o modules/detectron/CMakeFiles/caffe2_detectron_ops.dir/softmax_focal_loss_op.cc.o modules/detectron/CMakeFiles/caffe2_detectron_ops.dir/spatial_narrow_as_op.cc.o modules/detectron/CMakeFiles/caffe2_detectron_ops.dir/upsample_nearest_op.cc.o  -Wl,-rpath,/Users/thipok.tham/Documents/Projects/Misc/pytorch/build/lib  lib/libtorch.dylib  lib/libtorch_cpu.dylib  lib/libprotobuf.a  lib/libc10.dylib && :
Undefined symbols for architecture arm64:
  "___kmpc_for_static_fini", referenced from:
      _.omp_outlined. in upsample_nearest_op.cc.o
  "___kmpc_for_static_init_4", referenced from:
      _.omp_outlined. in upsample_nearest_op.cc.o
  "___kmpc_fork_call", referenced from:
      caffe2::UpsampleNearestOp<float, caffe2::CPUContext>::RunOnDevice() in upsample_nearest_op.cc.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```
Not sure if it is possible to skip this, it doesn't seems to be the core of pytorch

Anyway I tested on another environment install with `pip install torch torchvision torchaudio` which upgrade torch to `1.10.2`, which is now worked. So I tested again with `conda install pytorch torchvision -c pytorch`, but it produce the above error
